packages/iTwinUI-react/src/core/Table/filters/FilterToggle.tsx

```diff
@@ -53,13 +52,12 @@ export const FilterToggle = <T extends Record<string, unknown>>(
           placement='bottom-start'
           visible={isVisible}
           onClickOutside={close}
-          appendTo={ownerDocument?.body}
         >
           <IconButton
             styleType='borderless'
             isActive={isVisible || isColumnFiltered}
             className={cx('iui-filter-button', className)}
-            onClick={(e) => {
+            onMouseDown={(e) => {
```

Comment:
This will break for keyboard users because `click` automatically handles keyboard events too. Also it's a bit unexpected because click handlers usually activate when the mouse is released. 

Same comment in other places like BaseFilter

I think we should try to prevent the sort handler from being called when resizing.